### PR TITLE
Adds a resonance cascade

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

Adds a resonance cascade to the sm, I finally finished it #49244
Basically the sm makes a "tweaked" portal storm when it delams at maximum fuck.
I've added a couple of rarely used/thematic mobs to the list. The amount of prep required to pull this off means I am treating this as a shuttle call.

I also added a new mob, the headcrab. It turns any carbons it jumps on into zombies slowly, and can be shot/bashed off.
I'll add more detail and pics here in around 16 hours, I have nightcoded enough for this day.

## Why It's Good For The Game

Adds fun

## Changelog
🆑 LemonInTheDark, with thanks to OFQ for his sprites and anyone else who shot ideas my way.
add: Adds a new delam to the supermatter. Give it a shot some time.
/🆑

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
Miss ya man @JAremko

**Comment by Ghilker** [comment]
Half life in SS13!
giphy (1)

**Comment by tralezab** [comment]
Basically no good deed goes unpunished with my stupid comment but i feel like we should be inspired by other games, not directly ripping from them this hard. By putting mesa everywhere in our code and just straight up having headcrabs that take more damage from crowbars, you are not making for a universe people can discover and write lore for. you just straight uncreative. reminds me of when we had flans from final fantasy, and it was just egregiously obvious

still pretty cool though 👌

File: code/modules/events/portal_storm.dm
Comment (ATH1909) [inline code/modules/events/portal_storm.dm:31]: what about, the, ah, the creatures with flowers on their heads that shoot laser beams? should we add them to this list?

File: code/modules/power/supermatter/supermatter.dm
Comment (ATH1909) [inline code/modules/power/supermatter/supermatter.dm:679]: period

**Comment by ATH1909** [comment]
> Basically no good deed goes unpunished with my stupid comment but i feel like we should be inspired by other games, not directly ripping from them this hard. By putting mesa everywhere in our code and just straight up having headcrabs that take more damage from crowbars, you are not making for a universe people can discover and write lore for. you just straight uncreative. reminds me of when we had flans from final fantasy, and it was just egregiously obvious
> 
> still pretty cool though, if my biggest complaint is the theme i have very little to complain about 👌

Honestly, I think we could just replace the headcrabs with blob spores, and we'd probably be fine.

Can we allow all of these mobs to be possessed by ghosts, like spiders from a broodmother can? Maybe even add them to the list of ghost spawners?

File: code/modules/events/portal_storm.dm
Comment (spookydonut) [reply code/modules/events/portal_storm.dm:31]: what

Comment (trollbreeder) [reply code/modules/events/portal_storm.dm:31]: He means the Seedling.

File: code/modules/power/supermatter/supermatter.dm
Comment (trollbreeder) [reply code/modules/power/supermatter/supermatter.dm:679]: ATH please learn how to use code suggestions
```suggestion
					radio.talk_into(src, "Warning: Anti-mass spectrometer showing a five percent gain.", engineering_channel)
```

File: code/modules/events/portal_storm.dm
Comment (Fikou) [inline code/modules/events/portal_storm.dm:34]: larvas dont move

File: code/modules/events/portal_storm.dm
Comment (Fikou) [inline code/modules/events/portal_storm.dm:33]: why is a reskinned legion a boss

**Comment by shellspeed1** [comment]
Prepare for unforeseen consequences.

File: code/modules/events/portal_storm.dm
Comment (LemonInTheDark) [reply code/modules/events/portal_storm.dm:33]: Cause it's unused and "alien"

File: code/modules/events/portal_storm.dm
Comment (LemonInTheDark) [reply code/modules/events/portal_storm.dm:34]: it's mostly for fluff, I wanted to spook people a bit

File: code/modules/power/supermatter/supermatter.dm
Comment (LemonInTheDark) [reply code/modules/power/supermatter/supermatter.dm:679]: Can do

File: code/modules/events/portal_storm.dm
Comment (LemonInTheDark) [reply code/modules/events/portal_storm.dm:31]: Uhhh maybe, not too sure on what the mob is like

**Comment by LemonInTheDark** [comment]
> 
> 
> Basically no good deed goes unpunished with my stupid comment but i feel like we should be inspired by other games, not directly ripping from them this hard. By putting mesa everywhere in our code and just straight up having headcrabs that take more damage from crowbars, you are not making for a universe people can discover and write lore for. you just straight uncreative. reminds me of when we had flans from final fantasy, and it was just egregiously obvious
> 
> still pretty cool though, if my biggest complaint is the theme i have very little to complain about 👌

Yeah I get that. This project is the result of me taking ideas from players and attempting to code them. I've made the inspirations kinda obvious code side, but it shouldn't be overly clear in game, aside from the green tinted bolts.

I do agree on theming, the unique parts of the game are what makes it interesting lore wise. 
As for "you just straight uncreative.", you're right. It's a muscle I need to train more, at least in terms of things like this. I'm glad you like the pr :)  

As it relates to reskinning the headcrustations, I'd be up for that. Depends on how obvious this is.

**Comment by Capsandi** [comment]
Small tweak to give it more intriguing lore implications; have an announcement from Central command saying 'The super matter is letting of transdimensional radiation signals!', then launch a pod storm at the station **_FROM ANOTHER DIMENSION!_** which is very similar to a portal storm but you can see the delivery method before the enemies. 
Make the pod icons alien enough and your left with an unknown interdimensional faction wreaking havoc on a station in a different dimension because they got static from that station. Are they scared of what could be making noise on the station? Perhaps it's a brand of nanotrasen in a different dimension? Maybe the creatures are escaping a universe controlled by an unknown god of hiveminds and mind control?
Also i believe if you change the crabs to blob spores and they zombify anyone ghosts can control the zombies, just like when a blob makes spores.

**Comment by LemonInTheDark** [comment]
Excuse me? It works on my machine.

**Comment by jacklecroy** [comment]
Why have headcrabs if we have facehuggers already? There is lamarr, who is a headcrab in Half-Life 2, but is a debeaked facehugger in the game.

File: code/modules/events/portal_storm.dm
Comment (EgorDinamit) [reply code/modules/events/portal_storm.dm:34]: The bad part here is that some funny (TM) xenobiologist will give it a sentience potion.

**Review by Twaticus — CHANGES_REQUESTED** [review]
icons/obj/clothing/hats.dmi should have just 1 icon instead of all the directionals

File: code/modules/mob/living/simple_animal/hostile/headcrustation.dm
Comment (Rohesie) [inline code/modules/mob/living/simple_animal/hostile/headcrustation.dm:25]: This seems unused.

File: code/modules/mob/living/simple_animal/hostile/headcrustation.dm
Comment (Rohesie) [inline code/modules/mob/living/simple_animal/hostile/headcrustation.dm:37]: Nice meme. Use `wrench_act`

File: code/modules/power/supermatter/supermatter.dm
Comment (Rohesie) [inline code/modules/power/supermatter/supermatter.dm:334]: This implies the `supermatter_crystal` is null-spaced and likely deleted.
Either substitute this with `if(QDELETED(src))` or refactor it to guarantee it's not called if qdeleting.

File: code/modules/power/supermatter/supermatter.dm
Comment (Rohesie) [inline code/modules/power/supermatter/supermatter.dm:344]: Why are you sleeping here?

File: code/modules/projectiles/projectile/special/crab.dm
Comment (Rohesie) [inline code/modules/projectiles/projectile/special/crab.dm:74]: This will trigger even if equipped on hands  / grabbed.

File: code/modules/projectiles/projectile/special/crab.dm
Comment (Rohesie) [inline code/modules/projectiles/projectile/special/crab.dm:19]:
```suggestion
		if(istype(victim.head, /obj/item/clothing/head/headcrab))
```
`istype(null)` will return false already

File: code/modules/projectiles/projectile/special/crab.dm
Comment (Rohesie) [inline code/modules/projectiles/projectile/special/crab.dm:23]:
```suggestion
			if(H.is_mouth_covered(head_only = TRUE))
```
Boolean value, treat it as such.

File: code/modules/projectiles/projectile/special/crab.dm
Comment (Rohesie) [inline code/modules/projectiles/projectile/special/crab.dm:35]:
```suggestion
		victim.equip_to_slot_if_possible(creb, ITEM_SLOT_HEAD, FALSE, TRUE, TRUE)
```
Boolean vars, treat them as such.

File: code/modules/projectiles/projectile/special/crab.dm
Comment (Rohesie) [inline code/modules/projectiles/projectile/special/crab.dm:43]: These two procs can cause `death` to be called twice.

File: code/modules/projectiles/projectile/special/crab.dm
Comment (Rohesie) [inline code/modules/projectiles/projectile/special/crab.dm:50]:
```suggestion
	else if(iscarbon(hit))
```

File: code/modules/projectiles/projectile/special/crab.dm
Comment (Rohesie) [inline code/modules/projectiles/projectile/special/crab.dm:53]: Shouldn't this happen on item equip?

File: code/modules/mob/living/simple_animal/hostile/headcrustation.dm
Comment (LemonInTheDark) [reply code/modules/mob/living/simple_animal/hostile/headcrustation.dm:25]: Yeah I think I was trying to add possession support, but frankly I don't think I can properly.

File: code/modules/power/supermatter/supermatter.dm
Comment (LemonInTheDark) [reply code/modules/power/supermatter/supermatter.dm:334]: I'll replace it.

File: code/modules/power/supermatter/supermatter.dm
Comment (LemonInTheDark) [reply code/modules/power/supermatter/supermatter.dm:344]: the intent is to have a pause, but on reconsideration the portal storm isn't snappy enough for the effect I was going for.

File: code/modules/projectiles/projectile/special/crab.dm
Comment (LemonInTheDark) [reply code/modules/projectiles/projectile/special/crab.dm:43]: That shouldn't be possible, unless the crab had negative damage on the initial shoot. The math should be correct, not totally sure on that tho

File: code/modules/projectiles/projectile/special/crab.dm
Comment (LemonInTheDark) [reply code/modules/projectiles/projectile/special/crab.dm:35]: thank ya

File: code/modules/projectiles/projectile/special/crab.dm
Comment (LemonInTheDark) [reply code/modules/projectiles/projectile/special/crab.dm:23]: thank  ya

File: code/modules/projectiles/projectile/special/crab.dm
Comment (LemonInTheDark) [reply code/modules/projectiles/projectile/special/crab.dm:19]: aight

File: code/modules/projectiles/projectile/special/crab.dm
Comment (LemonInTheDark) [reply code/modules/projectiles/projectile/special/crab.dm:74]: I'm going to check the slot

File: code/modules/mob/living/simple_animal/hostile/headcrustation.dm
Comment (LemonInTheDark) [reply code/modules/mob/living/simple_animal/hostile/headcrustation.dm:37]: Arm was right.  Will do

File: code/modules/projectiles/projectile/special/crab.dm
Comment (LemonInTheDark) [reply code/modules/projectiles/projectile/special/crab.dm:50]: can do

**Comment by LemonInTheDark** [comment]
Should be handled @Twaticus

**Comment by nicbn** [comment]
I dislike it, it doesn't really contribute to the worldbuilding or anything, it's just a bare reference and that's all. May be cool if it's a random event, but inserting this as part of our universe seems... cheap.

File: code/modules/mob/living/simple_animal/hostile/headcrustation.dm
Comment (Rohesie) [inline code/modules/mob/living/simple_animal/hostile/headcrustation.dm:35]: This might not behave as you expect it to.
It will drain 10 integrity without an animation or anything, and then proceed to attack, causing the potential attack and damage of a dead mob. If it gibs/deletes that will cause issues.

**Comment by LemonInTheDark** [comment]
As it stands this is kinda a relic of a old rush of interest. I'm not so sure about it anymore.

File: code/modules/mob/living/simple_animal/hostile/headcrustation.dm
Comment (LemonInTheDark) [reply code/modules/mob/living/simple_animal/hostile/headcrustation.dm:35]: How would I go about resolving that?

Comment (Rohesie) [reply code/modules/mob/living/simple_animal/hostile/headcrustation.dm:35]: Well, you could The way other objects like `/obj/structure/spacevine` seems to be doing it is by redefining `attacked_by()` but this is old code, and not a great standard.
The ideal way would be to refactor `attacked_by()` to allow for special weaknesses when handling damage.

Comment (LemonInTheDark) [reply code/modules/mob/living/simple_animal/hostile/headcrustation.dm:35]: Ahhh, I'll look into it. May put this on hold, I'm not sure yet

Comment (Rohesie) [reply code/modules/mob/living/simple_animal/hostile/headcrustation.dm:35]: Take your time, poke if you need help.

**Comment by Acer202** [comment]
> As it stands this is kinda a relic of a old rush of interest. I'm not so sure about it anymore.

As a huge Half-Life fan, I was looking forward to it. I mean, we already have "borrowed" content, and a lot of people enjoy it. Seeing some headcrustacean zombies would be pretty crazy. Thanks for putting in the effort to make this in any case, and if it doesn't end up being something you want to continue it will inspire some creativity for the Singulam+Teslam combination result in other ways down the line anyways, I think.

**Comment by LemonInTheDark** [comment]
The reaper comes for us all. I'ma do a little tweaking to dial back on the half life, expect more when I can be bothered

**Comment by ArchieBeepBoop** [comment]
Did you ever end up adding a video of this? It sounds and looks pretty dope.

**Comment by LemonInTheDark** [comment]
nah, frankly the whole thing was kinda depressing. If you think it's neat feel free to port it, I spent a good bit of time on it and seeing it come to some use would be wonderful.

**Comment by ArchieBeepBoop** [comment]
Sometimes what you have to do is look for a coder base that is more welcoming to these sorts of changes. One with maintainers and other devs that will actually review, test and with game designers to provide proper feedback.

Personally I stray away from tg because of how tight the leash they keep on maintainers is.

**Comment by LemonInTheDark** [comment]
nah it's not like that, tg's been pretty good to me. I just went a bit too far with this pr, least in my opinion. it's like adding several large *insert popular show* refs at once. Feels soulless

**Pull request closed without merging**
